### PR TITLE
Allow larger values of MAX_MIXINPUTS

### PR DESCRIPTION
**Pull request by cdknox**
Removed the bit logic in favor of an array based approach to allow more inputs to a mixer than address size would allow.  

Certainly more comparisons, but I've been running it for a while with no issues it seems.

I appreciate the work!

### CODE REVIEW

File: src/mixer.cpp
Comment (charlie-foxtrot) [inline src/mixer.cpp:88]: looks like a mix of spacing with the rest of the file (tabs vs spaces) here and a few other places

File: src/mixer.cpp
Comment (charlie-foxtrot) [inline src/mixer.cpp:202]: this can also be replaced with a bool rather than counting

File: src/mixer.cpp
Comment (charlie-foxtrot) [inline src/mixer.cpp:203]: another binary and

File: src/mixer.cpp
Comment (charlie-foxtrot) [inline src/mixer.cpp:92]: I think this can go up to `mixer->input_count` rather than `MAX_MIXINPUTS` (check other places as well)

File: src/mixer.cpp
Comment (charlie-foxtrot) [inline src/mixer.cpp:208]: this seems less than ideal.  these debug printouts are already quite verbose, adding another line for each mixer input with the only difference being `mixer->inputs_todo[k]`, `mixer->input_mask[k]` is a lot more output.  How about generating a single string that has the list of true / false for each, and including that once?

**Review by charlie-foxtrot — CHANGES_REQUESTED** [review]
thanks for the change, left a few comments, and agree the performance impact shouldn't be a concern.

under the current model PRs are opened against the `unstable` branch, merged there, then `unstable` is merged to `main` on the tagging of a release.  I intend to change this, but for now I'll change your PR to be against `unstable`

**Comment by charlie-foxtrot** [comment]
@cdknox NOTE - I changed the PR baseline to `unstable` and clicked the button to merge `unstable` into this branch.  I'm not sure if that change will propagate into your fork, but you'll want to do a `git pull` before making changes otherwise you'll be on a divergent branch

**Comment by cdknox** [comment]
Fair points and thanks for the feedback!  I believe I addressed all of them now and it looks like I was able to get the rebasing done appropriately for the review (apologies for missing it should have been done against unstable).

Python is my native language so what I came up with for the formatting is not the most elegant I'm sure, but it does appear to be functional.  It's a string of 0s and 1s with the leftmost index being the 0th input and counting up from there.  I've included a couple examples at the bottom, one with 13 inputs and one with 47 (after having compiled with an increased MAX_MIXINPUTS of 64).

mixer_thread(): mixerinput: 1690435113.452584 27 int=1 inp_unhandled=0000000000000 inp_mask=1111111111111

mixer_thread(): mixerinput: 1690435113.452557 93 int=1 inp_unhandled=00000000000000000000000000000000000000000000000 inp_mask=11111111111111111111111111111111111111111111111

**Comment by charlie-foxtrot** [comment]
awesome, thanks!

There are a few more minor tweaks and rather than going back and forth I'll just make them.  If im unable to push to your fork I'll do it as a new PR.

**Comment by charlie-foxtrot** [comment]
@cdknox how high are you looking at setting `MAX_MIXINPUTS`?  This change will make the code more flexible so that larger values of `MAX_MIXINPUTS` can be used, but that still is a hard coded value in the code.  An alternative would be to change to a dynamic size list / vector so there is no hard coded limit at all. . . . Do you have a valid use case that requires more than 32?

**Comment by charlie-foxtrot** [comment]
there is a request to increase the value in #401

**Comment by cdknox** [comment]
@charlie-foxtrot  

re: the small changes, all good if you want to make them or I can if it'd make your life easier, I believe you should have access to commit to the branch, but if there's something setting I need to change otherwise just let me know.

re: `MAX_MIXINPUTS`, I had 47 inputs that were Indy Center being mixed together. That categorically ended up being too much to digest all at once, but I did like being able to just add them all and see immediately as opposed to trying to find freqs near me up to a limit.  In the case of #401 where it sounds like the maritime frequencies are very sparse I could understand chucking a lot into the mixer just in case some keyed up.  The way I got to 47 was to just edit the header file with `MAX_MIXINPUTS` in it and then compile but I could see that not being particularly user friendly.  As a low effort flexible option not requiring a lot of rework, I think it's possible to expose it as a build option through cmake (?) and just update the documentation.  My cmake game is pretty weak so maybe I'm wrong there.  All that's my two cents anyway, I'll obviously defer to you as the maintainer!

**Comment by charlie-foxtrot** [comment]
Opened a new PR based on these changes #408 that removes the upper bound, closing this PR

**Pull request closed without merging**
